docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,206 @@
+have access to [nested virtualization](https://en.wikipedia.org/wiki/Virtualization#Nested_virtualization).
+While it is possible to run EVE-OS in a virtual environment without nested virtualization
+enabled a lot of capabilities (e.g. running accelerated VMs) will be degraded.
+
+## Live vs installer EVE-OS images
+
+Since EVE-OS follows a very traditional boot process, its *live* image has to be
+available on a storage device that is accessible by the BIOS. When it is already
+deployed EVE-OS provides capabilities for managing its own live image. However,
+in order to bootstrap EVE-OS the initial live image has to be written to the
+storage device somehow.
+
+In a lot of situations, the storage device can be written to separately and then
+attached back to the edge node. This applies to both physical (e.g.  taking a live
+EVE-OS image, writing it onto a USB stick and booting off of that USB stick) and
+virtual (taking a live EVE-OS image and instructing your virtualization platform
+to use it as a hard drive) environments. However, there are times when this is
+either impossible (e.g. the only storage device is an eMMC soldered to the edge
+node's board) or impractical (the USB controller is so slow that booting off of
+the external USB stick take a long time). In cases like these we need an initial
+"installation" process. All that process does is it writes EVE-OS live image
+onto the storage device that can only be accessed by code running on the edge
+node itself. This is called an *installer* image of EVE-OS. It is a separate
+binary that you can get from the EVE-OS distribution.
+
+There's nothing special about the *installer* image of EVE-OS. In fact, under the
+hood, it is simply a live image of EVE-OS that runs a single application: a tiny
+script that writes a live image (and since installer image is a live image it
+effectively writes itself) onto the storage device that is inaccessible from
+the outside of the edge-node. The right way to think about installer image is
+that it is a live image that is only ever booted once on a given edge-node. Once
+installer image is done and shuts the edge node down the live image is available
+on the storage device and there's no need to run installer image anymore.
+
+In general, EVE-OS assumes a hands-free environment and can't rely on a human
+operator to configure initial settings "at the console". Given that the installer
+image is nothing but EVE-OS, it too, needs to be told ahead of time about any
+configuration necessary. To find out more how EVE-OS gets configured follow
+[configuration](CONFIG.md) documentation. For the rest of this document we will
+assume either config partition having required configuration or dynamic configuration
+done by supplying overrides to the boot process of the installer image. 
+
+## Serial vs. soft serial numbers
+
+EVE controller has to be able to recognize EVE for the first time. This is done
```

Comment:
Should preface this with text explaining that this is needed if registration (see REGISTRATION.md) is needed; if the device.cert.pem has been extracted out of band and registered in the controller this "first time" does not apply.

One way to word that would be to replace "first time" with "should it need to do registration (see REGISTRATION.md) on boot".

Reply:
fixed